torch/autograd/_functions/pointwise.py

```diff
@@ -331,93 +370,133 @@ class Remainder(_ConstantGrad):
-        self.scale = scale
-
-    def forward(self, add_tensor, mul_tensor1, mul_tensor2):
-        self.save_for_backward(mul_tensor1, mul_tensor2)
-        if self.inplace:
-            self.mark_dirty(add_tensor)
-            return add_tensor.addcmul_(self.scale, mul_tensor1, mul_tensor2)
+    @staticmethod
+    def forward(ctx, add_tensor, *args, **argsd):
+        inplace = argsd.get("inplace", False)
+        if len(args) == 3 and type(args[-1]) != bool:
+            scale, mul_tensor1, mul_tensor2 = args
+            ctx._have_scale = 1
+        elif len(args) == 3:
+            scale = 1.0
+            mul_tensor1, mul_tensor2, inplace = args
+            ctx._have_scale = 0
+        elif len(args) == 4:
+            scale, mul_tensor1, mul_tensor2, inplace = args
+            ctx._have_scale = 1
+        else:
+            scale = 1.0
+            mul_tensor1, mul_tensor2 = args
+            ctx._have_scale = 0
+        ctx._scale = scale
+        ctx.save_for_backward(mul_tensor1, mul_tensor2)
+        if inplace:
+            ctx.mark_dirty(add_tensor)
+            return add_tensor.addcmul_(scale, mul_tensor1, mul_tensor2)
         else:
-            return add_tensor.addcmul(self.scale, mul_tensor1, mul_tensor2)
+            return add_tensor.addcmul(scale, mul_tensor1, mul_tensor2)
 
-    def backward(self, grad_output):
+    @staticmethod
+    def backward(ctx, grad_output):
         grad_add = grad_mul1 = grad_mul2 = None
-        mul_tensor1, mul_tensor2 = self.saved_tensors
+        mul_tensor1, mul_tensor2 = ctx.saved_variables
 
-        if self.needs_input_grad[0]:
+        if ctx.needs_input_grad[0]:
             grad_add = grad_output
 
-        if self.needs_input_grad[1]:
-            grad_mul1 = grad_output.mul(mul_tensor2).mul(self.scale)
+        if ctx._have_scale:
+            assert not ctx.needs_input_grad[1], "Gradient of scale not implemented"
+
+        if ctx.needs_input_grad[1 + ctx._have_scale]:
+            grad_mul1 = grad_output.mul(mul_tensor2).mul(ctx._scale)
 
-        if self.needs_input_grad[2]:
-            grad_mul2 = grad_output.mul(mul_tensor1).mul(self.scale)
+        if ctx.needs_input_grad[2 + ctx._have_scale]:
+            grad_mul2 = grad_output.mul(mul_tensor1).mul(ctx._scale)
 
-        return grad_add, grad_mul1, grad_mul2
+        if ctx._have_scale:
+            return grad_add, None, grad_mul1, grad_mul2, None
+        else:
+            return grad_add, grad_mul1, grad_mul2, None
 
 
 class Addcdiv(InplaceFunction):
 
-    def __init__(self, scale=1, inplace=False):
-        super(Addcdiv, self).__init__(inplace)
-        self.scale = scale
-
-    def forward(self, add_tensor, div_tensor1, div_tensor2):
-        self.save_for_backward(div_tensor1, div_tensor2)
-        if self.inplace:
-            self.mark_dirty(add_tensor)
-            return add_tensor.addcdiv_(self.scale, div_tensor1, div_tensor2)
+    @staticmethod
+    def forward(ctx, add_tensor, *args, **argsd):
```

Comment:
can't you just make the signature `(ctx, add_tensor, div_tensor1, div_tensor2, scale=1)`? This would simplify the code a lot, and it's simple to modify the Variable code to call it like that